pkg/codegen/java/gen_program.go

```diff
@@ -229,33 +229,295 @@ func containConfigVariables(nodes []pcl.Node) bool {
 	return false
 }
 
-// genPreamble generates import statements, class definition and constructor.
+func cleanModule(module string) string {
+	if strings.Contains(module, "/") {
+		// if it is a function module
+		// e.g. index/getAvailabilityZones
+		moduleParts := strings.Split(module, "/")
+		// unless it is just a version e.g. core/v1
+		// then return it as is
+		if len(moduleParts) == 2 && strings.HasPrefix(moduleParts[1], "v") {
+			return module
+		}
+		return moduleParts[0]
+	}
+
+	return module
+}
+
+// Decides when to ignore a module in import definition
+func ignoreModule(module string) bool {
+	return module == "" || module == "index"
+}
+
+// Removes dashes and replaces slash with underscore
+func sanitizeImport(name string) string {
+	// e.g. azure-native becomes azurenative
+	withoutDash := strings.ReplaceAll(name, "-", "")
+	// e.g. kubernetes.core/v1 becomes kubernetes.core_v1
```

Comment:
Link to https://github.com/pulumi/pulumi-java/issues/381 TODO as this might change.

```
// TODO[pulumi/pulumi-java#381]
```

Reply:
Done